### DISCUSSION TITLE
What is user <anonymous> appearing in logs when the LAN is not whitelisted?

### DISCUSSION THREAD

**Discussion by wsw70** [authelia/authelia#6767] [Troubleshooting] [answered]
_(I was not sure whether to post the question here or in Caddy's forum)_

# Context 

I had a working Authelia setup (together with Caddy) where all non-local requests were going through authentication. Below is the typical **Caddy** setup

```
https://authelia.swtk.eu {
	reverse_proxy authelia:9091 {
		trusted_proxies private_ranges
	}
}

(authenticate-with-authelia) {
	@forceAuth {
		not remote_ip private_ranges
		}
	forward_auth @forceAuth authelia:9091 {
		uri /api/verify?rd=https://authelia.swtk.eu/
		copy_headers Remote-User Remote-Groups Remote-Name Remote-Email
		trusted_proxies private_ranges
	}
}
```

It worked fine. I then decided to remove the whitelisting on the local addresses because i) why not and ii) the `Remote-User` is useful in some applications. After the change almost everything worked fine but now I get a bunch of logs in **Authelia**'s logs (the services differ, though not all of them appear):

```
authelia-1  | time="2024-03-02T08:21:41+01:00" level=info msg="Access to https://unifi.swtk.eu/ (method GET) is not authorized to user <anonymous>, responding with status code 302 with location redirect to https://authelia.swtk.eu/?rd=https%3A%2F%2Funifi.swtk.eu%2F&rm=GET" method=GET path=/api/verify remote_ip=192.168.10.1
authelia-1  | time="2024-03-02T08:21:41+01:00" level=info msg="Access to https://unifi.swtk.eu/api/login (method POST) is not authorized to user <anonymous>, responding with status code 303 with location redirect to https://authelia.swtk.eu/?rd=https%3A%2F%2Funifi.swtk.eu%2Fapi%2Flogin&rm=POST" method=GET path=/api/verify remote_ip=192.168.10.1
authelia-1  | time="2024-03-02T08:21:41+01:00" level=info msg="Access to https://unifi.swtk.eu/wss/s/8bjv5hqb/events (method GET) is not authorized to user <anonymous>, responding with status code 302 with location redirect to https://authelia.swtk.eu/?rd=https%3A%2F%2Funifi.swtk.eu%2Fwss%2Fs%2F8bjv5hqb%2Fevents&rm=GET" method=GET path=/api/verify remote_ip=192.168.10.1
authelia-1  | time="2024-03-02T08:42:02+01:00" level=info msg="Access to https://pihole.swtk.eu/admin/api.php?summaryRaw=&auth=None (method GET) is not authorized to user <anonymous>, responding with status code 302 with location redirect to https://authelia.swtk.eu/?rd=https%3A%2F%2Fpihole.swtk.eu%2Fadmin%2Fapi.php%3FsummaryRaw%3D%26auth%3DNone&rm=GET" method=GET path=/api/verify remote_ip=192.168.10.1
```

`192.168.10.1` is the gateway (ISP router).

# Troubleshooting

What is the `<anonymous>` user that Authelia refuses to connect?

I guess that it appears in a situation where Authelia gets redirected to outside of the normal login flow. How could this happen?
Or in other words: what kind of activity of an application do you think could trigger such logs?

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

The `<anonymous>` user mentioned in your Authelia logs indicates that the requests to your protected services (like `unifi.swtk.eu` and `pihole.swtk.eu`) are being made without any authenticated user context. This typically happens when a request reaches Authelia without a valid session cookie or without proper authentication headers, leading Authelia to treat the request as coming from an unauthenticated (or anonymous) user.

Given the context of your setup and the changes you made, there are a few potential reasons why you might be seeing these logs:

1. **Direct Access by Unauthenticated Users**: If your services are being accessed directly by users (or bots) without going through the authentication flow, Authelia will log these attempts as it redirects them to the login page. This could be common for services like UniFi or Pi-hole, which might be probed by automated scanners or accessed by users who haven't authenticated yet.

2. **Service-Specific Traffic**: Some services generate background network traffic (API calls, WebSocket connections, etc.) that might not carry the authentication cookies or headers correctly, especially when initiated by the service's web interface in the user's browser. This could explain why specific endpoints like `/api/login` or `/wss/s/.../events` are mentioned.

3. **Misconfiguration or Missing Headers**: Ensure that your Caddy configuration correctly forwards necessary headers to Authelia and that your services are correctly interpreting forwarded headers. Missing or misconfigured headers could lead to Authelia not recognizing authenticated sessions.

4. **Network Configuration**: The fact that `192.168.10.1` (your gateway) is mentioned might indicate that requests are being routed in a way that strips them of their authenticated context or that these requests originate from the gateway itself, possibly due to NAT configurations or other network behaviors.

To troubleshoot further, consider the following steps:

- **Review Service Logs**: Check the logs of the services generating these logs (UniFi, Pi-hole, etc.) to see if there's any indication of why these requests are being made without authentication.
- **Check Network Traffic**: Use tools like Wireshark to capture the network traffic to/from your services to understand better how these requests are being made and why they might lack authentication.
- **Review Authelia and Caddy Configurations**: Double-check your Authelia and Caddy configurations to ensure that all necessary headers are being forwarded correctly and that there are no misconfigurations that could lead to these issues.
- **Test with Known Good Configurations**: If possible, temporarily revert to a known good configuration (before your changes) to see if the issue persists, which could help isolate the cause.

Given the complexity of these systems and the potential for subtle configuration issues, it might take some investigation to pinpoint the exact cause. However, focusing on how authentication data is carried and preserved across your requests should help narrow down the issue.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment, marked as answer]
The `<anonymous>` is the name of an unauthenticated user, until they authenticate that's who they're known as, when they're not authenticated they're redirected to authenticate.

**Reply by wsw70** [reply]
Thank you. So that would mean that I have always had these logs and that they are not related to my changes (except that now that will also be triggered for a LAN first connection)? Unfortunately the earliest logs I have are post-change in the configuration.

**Reply by james-d-elliott** [reply]
Potentially, if you're whitelisting in some way based on IP it may not show them unless you're using a bypass rule in authelia in which case the logs will indicate the request was permitted.
